bugrap/src/main/java/org/vaadin/bugrap/ui/ReportDetailView.java

```diff
@@ -55,119 +55,122 @@ private void initializeUIComponents() {
 		btnRevertReport.addClickListener(evt -> revertChanges());
 		btnDone.addClickListener(evt -> saveComment());
 		btnCancel.addClickListener(evt -> closeWindow());
-		
 		btnUpload.setReceiver(this);
 		btnUpload.addStartedListener(this);
 		btnUpload.addSucceededListener(this);
 		btnUpload.addProgressListener(this);
+		txtComment.addValueChangeListener(evt -> commentsUpdated());
 	}
 
 	private void init() {
-		((HorizontalLayout)pnlAttachments.getContent()).removeAllComponents();
-		pnlAttachments.setVisible(false);
 		Report report = model.getReportForEdit();
-		binder.setBean(model.getReportForEdit());
+		binder.setBean(report);
 		lblProjectSummary.setValue(report.getSummary());
 		initializeComboContents();
 		fillComments();
+		cleanAttachments();
+	}
+
+	private void cleanAttachments() {
+		((HorizontalLayout) pnlAttachments.getContent()).removeAllComponents();
+		pnlAttachments.setVisible(false);
 	}
-	
+
 	private void initializeComboContents() {
 		cmbEditPriority.setItems(model.getPriorties());
 		cmbEditType.setItems(model.getTypes());
 		cmbEditStatus.setItems(model.getStatuses());
 		cmbEditAssignedTo.setItems(model.findReporters());
-		cmbEditVersion.setItems(model.findProjectVersions(model.getReportForEdit().getProject()));
+		cmbEditVersion.setItems(model.findProjectVersions());
 	}
-	
+
 	private void fillComments() {
 		VerticalLayout layout = (VerticalLayout) pnlCommentsThread.getContent();
 		layout.removeAllComponents();
 		List<Comment> comments = model.getComments();
 		for (Comment comment : comments) {
 			layout.addComponent(new ThreadItemView(new ThreadItemModel(comment)));
 		}
+		txtComment.clear();
+		btnDone.setEnabled(false);
 	}
-	
-	
+
+	private void commentsUpdated() {
+		btnDone.setEnabled(true);
+	}
+
 	private void saveReport() {
 		model.save();
 		Notification.show("Saved successfully", Type.TRAY_NOTIFICATION);
 	}
-	
+
 	private void revertChanges() {
 		model.resetReportForEdit();
 		binder.setBean(model.getReportForEdit());
 	}
 
 	private void saveComment() {
-		if (!txtComment.getOptionalValue().isPresent())
-			return;
-		model.saveComment(txtComment.getValue(), BaseModel.loginUser);
+		if (!txtComment.isEmpty())
+			model.saveComment(txtComment.getValue(), BaseModel.loginUser);
+		model.saveAttachments();
```

Comment:
What happens if there is a attachment still being uploaded when the user clicks Save?

Reply:
done button is disabled when upload starts